### PR TITLE
refactor: extract PathSharder module for shared multi-path selection

### PR DESCRIPTION
**Pull request by glimchb**
Move the duplicated path-sharding logic (CSV parsing, device ID extraction, by_gpu strategy) from LocalDiskBackend and GdsBackend into a new PathSharder class in path_sharder.py.

Both backends now delegate to PathSharder, keeping the policy in one place.  The _resolve_device_id helper handles cuda:N, bare cuda, and cpu devices gracefully.

Generated with [Devin](https://cli.devin.ai/docs)

**What this PR does / why we need it**:

**Special notes for your reviewers**:

**If applicable**:

- [ ] this PR contains user facing changes - docs added
- [ ] this PR contains unit tests

---

> [!NOTE]
> **Medium Risk**
> Touches storage backend initialization and path selection logic for both `LocalDiskBackend` and `GdsBackend`, which could change which directory is used (and how invalid configs fail) in multi-GPU/multi-path deployments.
> 
> **Overview**
> Refactors multi-path cache directory selection by extracting the shared CSV parsing + *by-GPU* sharding logic into a new `PathSharder` utility and switching both `GdsBackend` and `LocalDiskBackend` to use it.
> 
> Standardizes device-id resolution (handles `cuda:N`, bare `cuda`, and `cpu`) and changes unsupported sharding configuration failures from `AssertionError` to `ValueError`, with updated/new unit tests covering selection, directory creation, and edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 192341413dca14d16fb3a44e510d65d2961f78cf. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by DongDongJu** [comment]
Hello @glimchb, Thanks for the hard work. Could you check bot review?
Otherwise, functionally LGTM!

File: lmcache/v1/storage_backend/path_sharder.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    Returns:
+        Integer device index.  Falls back to
+        :func:`torch.cuda.current_device` when the string carries no
+        explicit index, or ``0`` when CUDA is unavailable.
+    """
+    if ":" in dst_device:
+        return int(dst_device.split(":")[1])
+    if torch.cuda.is_available():
+        return torch.cuda.current_device()
+    return 0
```
[the remark this answers is not in the document]
Comment (glimchb) [reply lmcache/v1/storage_backend/path_sharder.py:42]: done

File: tests/v1/storage_backend/test_gds_backend.py [hunk 2]
```diff
@@ -763,10 +763,10 @@ def test_try_to_read_metadata_finds_across_all_paths(self, async_loop):
                 shutil.rmtree(p, ignore_errors=True)
 
     def test_gds_path_sharding_default(self, temp_gds_path, async_loop):
-        """Default gds_path_sharding is 'by_gpu'."""
+        """Default gds_path_sharding is 'by_gpu' (backend inits OK)."""
```
Comment (DongDongJu) [inline tests/v1/storage_backend/test_gds_backend.py:766]: What does "backend inits OK" meaning?

**Review by DongDongJu — APPROVED** [review]
LGTM.
I left one nit question.

**Review by sammshen — APPROVED** [review]
LGTM
